planner/core/rule_aggregation_elimination.go

```diff
@@ -116,13 +116,25 @@ func (a *aggregationEliminateChecker) wrapCastFunction(ctx sessionctx.Context, a
 	return expression.BuildCastFunction(ctx, arg, targetTp)
 }
 
-func (a *aggregationEliminator) optimize(p LogicalPlan) (LogicalPlan, error) {
+func (a *aggregationEliminator) optimize(p LogicalPlan) (_ LogicalPlan, err error) {
+	_, isProj := p.(*LogicalProjection)
+	needProjEliminate := false
 	newChildren := make([]LogicalPlan, 0, len(p.Children()))
 	for _, child := range p.Children() {
 		newChild, _ := a.optimize(child)
+		if isProj {
+			_, isOldChildAgg := child.(*LogicalAggregation)
+			_, isNewChildProj := newChild.(*LogicalProjection)
+			if isOldChildAgg && isNewChildProj {
+				needProjEliminate = true
+			}
+		}
 		newChildren = append(newChildren, newChild)
 	}
 	p.SetChildren(newChildren...)
+	if needProjEliminate {
+		return (&projectionEliminater{}).optimize(p)
```

Comment:
In https://github.com/pingcap/tidb/pull/7531/files#diff-6a7fc6244769b48bc271de3d766c529eR37, we introduce the `flagEliminateProjection2`, could this be done by setting that flag? Or we may eliminate the projection multiply times.